docs/placeholder.md

```diff
@@ -1,2 +1,380 @@
+
+## Prerequisites
+
+### Environment Setup
+
+```bash
+# Clone the repository
+git clone <deafrica_water_quality>
+
+# Install in editable mode
+cd deafrica_water_quality
+pip install -e .
+
+# Restart kernel if working on DE Africa Sandbox
+```
+
+### Requirements
+- Access to DE Africa Sandbox datacube environment
+- Python environment with required dependencies
+- YAML configuration file for analysis parameters
+- Appropriate filesystem access (local or cloud storage)
+
+---
+
+## Pipeline Architecture
+
+### Stage 1: Task Generation (`wq-generate-tasks`)
+
+Generates a list of processing tasks based on temporal range, frequency, and spatial tiles.
+
+#### Command
+
+```bash
+!wq-generate-tasks [OPTIONS] TEMPORAL_RANGE FREQUENCY OUTPUT_FILE
+```
+
+#### Arguments
+- **TEMPORAL_RANGE**: Time period in ISO format (e.g., `2020-05--P1M` for May 2020)
+- **FREQUENCY**: Temporal binning interval
+  - Supported: `annual`, `semiannual`, `monthly`, `fortnightly`, `weekly`
+- **OUTPUT_FILE**: Path to write task list (text file with one task per line)
+
+#### Options
+- `--tile-ids`: Comma-separated tile IDs (e.g., `x188/y109,x178/y095`)
+- `--tile-ids-file`: Path to file containing tile IDs
+- `--place-name`: Predefined test area name (view with `wq-list-test-areas`)
+
+**Note**: Specify exactly one of `--tile-ids`, `--tile-ids-file`, or `--place-name`. If none specified, generates tasks for all Africa tiles.
+
+#### Temporal Range Behavior
+If the temporal range doesn't completely cover a temporal bin, all data for the complete bin will be loaded:
+- `2025-08-13--P3D` with `weekly` frequency → loads `2025-08-13--P1W`
+- `2025-08-13--P1M` with `monthly` frequency → loads `2025-08--P1M` and `2025-09--P1M`
+
+#### Output
+Text file containing task IDs in format: `{temporal_id}/x{xx}/y{yy}`
+
+Example: `2015--P1Y/x200/y034`
+
+---
+
+### Stage 2: Water Quality Processing (`wq-process-annual-wq-variables`)
+
+Processes tasks to generate water quality variables and outputs as Cloud Optimized GeoTIFFs (COGs).
+
+#### Command
+
+```bash
+!wq-process-annual-wq-variables [OPTIONS] OUTPUT_DIRECTORY MAX_PARALLEL_STEPS WORKER_IDX
+```
+
+#### Arguments
+- **OUTPUT_DIRECTORY**: Directory for output COG files, CSVs, and STAC metadata
+- **MAX_PARALLEL_STEPS**: Total number of parallel workers/pods in the workflow
+- **WORKER_IDX**: Zero-indexed worker ID (determines which task subset to process)
+
+#### Options
+- `--tasks`: Comma-separated task IDs
+- `--tasks-file`: Path to task file (generated by Stage 1)
+- `--analysis-config`: **Required** YAML configuration file path
+- `--overwrite / --no-overwrite`: Reprocess existing tasks (default: False)
+
+**Note**: Must specify either `--tasks` or `--tasks-file`, but not both.
+
+---
+
+## Configuration File
+
+### Example `cfg.yaml`
+
+```yaml
+resolution: 30  # Spatial resolution in meters
+
+# Satellite instruments to use
+instruments_to_use:
+  oli_agm:    # Landsat 8/9 OLI with atmospheric correction
```

Comment:
The difference between `oli` and `oli_agm` is any instrument with _agm is an annual GeoMAD.